optimum/onnx/configuration.py

```diff
@@ -147,52 +177,21 @@ def generate_dummy_inputs(
-                    seq_length, fixed_dimension=self.default_fixed_sequence, num_token_to_add=0
-                )
-                if framework == TensorType.PYTORCH:
-                    if not is_torch_available():
-                        raise ValueError(
-                            "Could not generate dummy inputs to perform the ONNX export through PyTorch because not installation was found."
-                        )
-                    else:
-                        import torch
-
-                        dummy_inputs[label] = torch.zeros(batch_size, seq_length, dtype=torch.long)
-                elif framework == TensorType.TENSORFLOW:
-                    if not is_tf_available():
-                        raise ValueError(
-                            "Could not generate dummy inputs to perform the ONNX export through TensorFlow because not installation was found."
-                        )
-                    else:
-                        import tensorflow as tf
-
-                        dummy_inputs[label] = tf.zeros(batch_size, seq_length, dtype=tf.int64)
-            else:
-                if framework == TensorType.PYTORCH:
-                    if not is_torch_available():
-                        raise ValueError(
-                            "Could not generate dummy inputs to perform the ONNX export through PyTorch because not installation was found."
-                        )
-                    else:
-                        import torch
-
-                        dummy_inputs[label] = torch.zeros(batch_size, dtype=torch.long)
-                elif framework == TensorType.TENSORFLOW:
-                    if not is_tf_available():
-                        raise ValueError(
-                            "Could not generate dummy inputs to perform the ONNX export through TensorFlow because not installation was found."
-                        )
-                    else:
-                        import tensorflow as tf
+        label_batch_size = compute_effective_axis_dimension(
+            batch_size, fixed_dimension=self.default_fixed_batch, num_token_to_add=0
+        )
+        label_seq_length = compute_effective_axis_dimension(
+            seq_length, fixed_dimension=self.default_fixed_sequence, num_token_to_add=0
+        )
 
-                        dummy_inputs[label] = tf.zeros(batch_size, dtype=tf.int64)
-        return dummy_inputs
+        if framework == TensorType.PYTORCH and is_torch_available():
+            return self._generate_extra_dummy_inputs_pt(dummy_inputs, label_batch_size, label_seq_length)
+        elif framework == TensorType.TENSORFLOW and is_tf_available():
+            return self._generate_extra_dummy_inputs_tf(dummy_inputs, label_batch_size, label_seq_length)
+        else:
+            raise ValueError(
+                f"Could not generate dummy inputs to perform the ONNX export through {framework} because not installation was found."
```

Comment:
Something that could be interesting in order to better target the error would be to add some function such [as](https://github.com/huggingface/transformers/blob/main/src/transformers/onnx/features.py#L351)

Reply:
Good point! Thanks Ella, more precise framework information is updated with adding the consideration of frameworks other than PyTorch and TensorFlow. @echarlaix